From: Ben Hutchings <ben@decadent.org.uk>
Subject: user_ns: Avoid ABI change in 3.10.6
Date: Sat, 17 Aug 2013 01:47:55 +0200
Forwarded: not-needed

commit 5c5f9cd2b539 ('userns: limit the maximum depth of
user_namespace->parent chain', commit 8742f229b635 upstream)
added a new field to struct user_namespace.

struct user_namespace is only allocated by the core kernel, so move
the new field to the end and hide it from genksyms.

--- a/include/linux/user_namespace.h
+++ b/include/linux/user_namespace.h
@@ -23,12 +23,14 @@ struct user_namespace {
 	struct uid_gid_map	projid_map;
 	atomic_t		count;
 	struct user_namespace	*parent;
-	int			level;
 	kuid_t			owner;
 	kgid_t			group;
 	unsigned int		proc_inum;
 	bool			may_mount_sysfs;
 	bool			may_mount_proc;
+#ifndef __GENKSYMS__
+	int			level;
+#endif
 };
 
 extern struct user_namespace init_user_ns;
